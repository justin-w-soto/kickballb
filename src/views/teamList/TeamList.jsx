import React from 'react'
import {useEffect, useState} from 'react'
import { Link } from 'react-router-dom';
import { deleteTeamById, getTeams } from '../../services/teams';

export const TeamList = () => {
    const [teams, setTeams] = useState([]);
 
    const loadTeams = async () => {
        const res = await getTeams()
        setTeams(res)
    }

    useEffect(() => {loadTeams()},[])
    
    const handleDelete = async ({id, name}) => {
        const shouldDelete = window.confirm(`Are you sure you want to delete the ${name} from the roster?`)
        
        if (shouldDelete) await deleteTeamById(id)
        await loadTeams()
    }

    return (
        <div>
            <h1>Teams</h1>
            <Link to='/teams/new' className='App-link'>Add a Team</Link>
            <ul>
            {teams.map((team) => (
                <li key={team.id}>

                    <Link className='link' to={`/teams/${team.id}`}>
                        <li>{team.name}</li>
                    </Link>

                    <button type="button">Edit</button>
                    <button 
                    type="button" aria-label={`Delete ${team.name}`}
                    onClick={() => handleDelete({ id: team.id, name: team.name })}>Delete</button>
                </li>
            ))}
            </ul>
        </div>
    )
}
