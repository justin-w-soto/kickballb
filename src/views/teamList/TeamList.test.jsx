import { render, screen } from '@testing-library/react';
import { TeamList } from './TeamList';
import { setupServer } from 'msw/node'
import { rest } from 'msw'
import { Route, MemoryRouter } from 'react-router-dom'
import userEvent from '@testing-library/react'

const mockTeam1 = {
  "id":1,
  "created_at":"2021-12-10T05:58:53.980677+00:00",
  "name":"Hero Ballers From OuterSpace",
  "city":"Kent",
  "state":"WA",
  "players": [],
}

const mockTeam2 = {
  "id":2,
  "created_at":"2021-12-10T05:58:53.980677+00:00",
  "name":"Magic Mazoos",
  "city":"Kalamazoo",
  "state":"MI",
  "players": [],
}

const server = setupServer(
  rest.get('https://pzqhqmkdmqivtsvqgatr.supabase.co/rest/v1/teams', (req, res, ctx) => {
    return res(ctx.json([mockTeam1, mockTeam2]))
}))

  rest.delete('https://pzqhqmkdmqivtsvqgatr.supabase.co/rest/v1/teams', (req, res, ctx) => {
      return res(ctx.json(mockTeam2))
  })

  beforeAll(() => {
    server.listen()
})

afterAll(() => {
    server.close()
})


it('should render a TeamList page', async () => {
  const { container } = render(
    <MemoryRouter>  
      <TeamList /> 
    </MemoryRouter>
  );
  await screen.findByText(/Hero Ballers From OuterSpace/i)
  expect(container).toMatchSnapshot();
});

it('should delete a team from the list', async () => {
  render(
    <MemoryRouter>
      <TeamList />
    </MemoryRouter>
  )
  await screen.findByText(/Magic Mazoos/i)
  const deleteBtn = screen.getByRole('button', { name: 'Delete Magic Mazoos' });

  // userEvent.click(deleteBtn)
})

