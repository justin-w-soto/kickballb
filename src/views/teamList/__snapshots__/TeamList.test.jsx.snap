// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render a TeamList page 1`] = `
<div>
  <div>
    <h1>
      Teams
    </h1>
    <a
      class="App-link"
      href="/teams/new"
    >
      Add a Team
    </a>
    <ul>
      <li>
        <a
          class="link"
          href="/teams/1"
        >
          <li>
            Hero Ballers From OuterSpace
          </li>
        </a>
        <button
          type="button"
        >
          Edit
        </button>
        <button
          aria-label="Delete Hero Ballers From OuterSpace"
          type="button"
        >
          Delete
        </button>
      </li>
      <li>
        <a
          class="link"
          href="/teams/2"
        >
          <li>
            Magic Mazoos
          </li>
        </a>
        <button
          type="button"
        >
          Edit
        </button>
        <button
          aria-label="Delete Magic Mazoos"
          type="button"
        >
          Delete
        </button>
      </li>
    </ul>
  </div>
</div>
`;
